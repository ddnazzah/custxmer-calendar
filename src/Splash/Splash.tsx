/* eslint-disable react-hooks/exhaustive-deps */
/*
 * @Author: Dieu-Donne Nazzah
 * @Date: 2022-10-24 04:39:53
 * @Last Modified by: Dieu-Donne Nazzah
 * @Last Modified time: 2022-10-25 18:12:34
 */
import { useNavigation } from '@react-navigation/native';
import { Constants } from '@shared/constants';
import { useSettings } from '@shared/hooks';
import React, { useEffect } from 'react';
import { StatusBar } from 'react-native';
import {
    CalendarText,
    Container,
    CustxmerText,
} from './components/styledComponents';

const Splash = () => {
    const { _theme } = useSettings();
    const { navigate } = useNavigation<any>();

    useEffect(() => {
        setTimeout(() => {
            navigate(Constants.routes.CALENDAR);
        }, 2000);

        return () => {};
    }, []);

    return (
        <Container colors={['#000000', _theme.PRIMARY, '#111CAF']}>
            <StatusBar hidden />
            <CustxmerText>Custxmer</CustxmerText>
            <CalendarText>Calendar</CalendarText>
        </Container>
    );
};

export default Splash;
