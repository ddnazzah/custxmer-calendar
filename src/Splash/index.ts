/*
 * @Author: Dieu-Donne Nazzah
 * @Date: 2022-10-24 04:02:16
 * @Last Modified by: Dieu-Donne Nazzah
 * @Last Modified time: 2022-10-24 04:39:18
 */

export { default as Splash } from './Splash';
