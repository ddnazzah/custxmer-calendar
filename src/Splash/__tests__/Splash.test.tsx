/*
 * @Author: Dieu-Donne Nazzah
 * @Date: 2022-10-27 18:08:55
 * @Last Modified by: Dieu-Donne Nazzah
 * @Last Modified time: 2022-10-27 19:02:33
 */

import { render, screen } from '@testing-library/react-native';
import React from 'react';
import TestProvider from '../../@testing/Provider';
import Splash from '../Splash';

describe('Splash', () => {
    it('renders correctly', () => {
        render(
            <TestProvider>
                <Splash />
            </TestProvider>,
        );

        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders text correctly', () => {
        render(
            <TestProvider>
                <Splash />
            </TestProvider>,
        );

        expect(screen.getByText('Custxmer')).toBeDefined();
        expect(screen.getByText('Calendar')).toBeDefined();
    });

    // it('navigates to MyCalendar', () => {
    //     render(
    //         <TestProvider>
    //             <Splash />
    //         </TestProvider>,
    //     );

    //     sleep(3);

    //     expect(screen.getByTestId('myCalendar')).toBeDefined();
    // });
});
