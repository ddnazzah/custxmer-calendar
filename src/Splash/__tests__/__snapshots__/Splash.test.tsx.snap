// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4278190080,
      4294400067,
      4279311535,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 36,
          "letterSpacing": 1,
        },
      ]
    }
  >
    Custxmer
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 36,
          "letterSpacing": 1,
        },
      ]
    }
  >
    Calendar
  </Text>
</BVLinearGradient>
`;
